### ISSUE TITLE
[code] 'convert(Vector{Char},delim)' make MethodError for julia 1.4

### ISSUE THREAD

**Issue by touchft** [JuliaIO/LibSerialPort.jl#66]
The following function in **high-level-api.jl** contains the code `convert(Vector{Char},delim)` which introduce an ERROR!
```
function Base.readuntil(sp::SerialPort, delim::AbstractString, timeout_ms::Real)
    return readuntil(sp,convert(Vector{Char},delim),timeout_ms)
end
```
julia> convert(Vector{Char},"ab")
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Array{Char,1}

It seems correct if we change the above code `convert(Vector{Char},delim)` to `Vector{Char}(delim)` .

julia> Vector{Char}("ab")
2-element Array{Char,1}:
 'a'
 'b'

**Comment by andrewadare** [comment]
PR #61 removes this Base.readuntil definition. Is it possible for you to try that branch and see if things work better?

**Comment by touchft** [comment]
I've tested PR #61 . I meet some strange problems. I've made couple tests:

1. send binary data (such .jpg ) to serial A Tx, which connect to serial B Rx
2. send binary data (such .jpg ) to serial A Tx, which connect to serial A Rx

The function `read()` or `nonblocking_read()` from test 1 or test 2 RX port will lost one byte every 1023 bytes received.  

For example, If I send 1024 bytes (raw) to Tx, I got 1023 bytes from Rx port.
If I send 2048 bytes to Tx, I got 2046 bytes from Rx port.

If I send a pure text file (all ASCII)  to Tx, the above problem is gone!

**Comment by andrewadare** [comment]
Apparently with PR 61 you do not have the problem that caused you to open this issue, which is good (and expected). 

Your new issue is not one that I can easily reproduce. It is unclear that this package is the cause. 

If you are determined to send image files over a serial connection, and you have observed that ASCII works, you might consider trying Base64 encoding, at least as a troubleshooting step.

**Comment by mgkuhn** [comment]
It seems all problems in this issue are already addressed in #61, including the fact that on some systems XON/XOFF is on by default and thus bytes 0x11 or 0x13 are swallowed. We need to make sure, in either the documentation or in the open() method, that users are likely to switch off control flow methods unless they actually know which ones their connected hardware actually supports. Encouraging the user of the high-level API to leave the operating-system default of flow control encourages non-portable programming, as different operating systems may have different defaults.

**Comment by mgkuhn** [comment]
Could you please close this issue?
